pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -2146,15 +2151,20 @@ func (cc *XController) Cleanup(ctx context.Context, appwrapper *arbv1.AppWrapper
 }
 func (cc *XController) getAppWrapper(namespace string, name string, caller string) (*arbv1.AppWrapper, error) {
 	klog.V(5).Infof("[getAppWrapper] getting a copy of '%s/%s' when called by '%s'.", namespace, name, caller)
-	apiCacheAWJob, err := cc.appWrapperLister.AppWrappers(namespace).Get(name)
-	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			klog.Errorf("[getAppWrapper] getting a copy of '%s/%s' failed, when called  by '%s', err=%v", namespace, name, caller, err)
+	var apiCacheAWJob *arbv1.AppWrapper
+	var err error
+	if cc.appwrapperInformer.Informer().HasSynced() {
```

Comment:
why is this change necessary ? Isn't the informer being synched in the `Run()` function?

Reply:
Thanks for pointing this out, will check the run function and make the necessary changes.

Reply:
Done.